### PR TITLE
rhel8 support and fix fedora 28+ support

### PR DESCRIPTION
**Pull request by robertc99**
This adds rhel8 support.
Also fixes the fedora 28+ support for mkhomedir.

### CODE REVIEW

File: REFERENCE.md [hunk 1]
```diff
@@ -143,49 +143,67 @@ service to prevent `Error: Could not start Service[oddjobd]`.
 
 Default value: []
 
-##### `enable_mkhomedir_flags`
+##### `authconfig_enable_mkhomedir_flags`
```
Comment (ghoneycutt) [inline REFERENCE.md]: why renaming all these parameters? this makes it a non-backward compatible change.

**Comment by robertc99** [comment]
Second attempt at rhel8 support.
Leave out the default parameter fixes.
API unchanged except for added authselect_profile parameter with sensible default so it won't break existing config.

File: REFERENCE.md (hunk 1 above)
Comment (robertc99) [reply REFERENCE.md]: New version resolves this issue.

**Comment by robertc99** [comment]
Second attempt at rhel8 support.
Leave out the default parameter fixes.
API unchanged except for added authselect_profile parameter with sensible default so it won't break existing config.

**Pull request closed without merging**